crates/solvers/src/domain/solver/dex.rs

```diff
@@ -0,0 +1,77 @@
+//! A simple solver that matches orders directly with swaps from the external
+//! DEX and DEX aggregator APIs.
+
+use {
+    crate::{
+        domain::{
+            auction,
+            dex::{self, slippage},
+            order, solution,
+        },
+        infra,
+    },
+    tracing::Instrument,
+};
+
+pub struct Dex {
+    /// The DEX API client.
+    pub dex: infra::dex::Dex,
+
+    /// The slippage configuration to use for the solver.
+    pub slippage: slippage::Limits,
+}
+
+impl Dex {
+    pub async fn solve(&self, auction: auction::Auction) -> Vec<solution::Solution> {
+        // TODO: order prioritization, skip liquidity orders, concurrency.
+        let prices = slippage::Prices::for_auction(&auction);
+
+        let mut solutions = Vec::new();
+        for order in auction.orders {
+            let span = tracing::info_span!("solve", order = %order.uid);
+            if let Some(solution) = self
+                .solve_order(order, &prices, auction.gas_price)
+                .instrument(span)
+                .await
+            {
+                solutions.push(solution);
+            }
+        }
+
+        solutions
+    }
+
+    async fn solve_order(
+        &self,
+        order: order::Order,
+        prices: &slippage::Prices,
+        gas: auction::GasPrice,
+    ) -> Option<solution::Solution> {
+        let query = dex::Order::new(&order);
+        let slippage = self.slippage.relative(&query.amount(), prices);
```

Comment:
Should we also check that the absolute slippage gets respected here or does that already happen at a higher level?